# Algortihm Parameters
ML:
  seed: 0
  obs_dim: 74
  n_act: 6
  epochs: 30
  hdim: 64
  policy_lr: 1e-5
  value_lr: 1e-5
  max_std: 1.0
  clip_range: 0.2
  n_step: 128 # 1024

  gamma: 0.99
  lam: 0.98
  episode_size: 1
  batch_size: 128
  nupdates: 30000

  act_step: 30 #100
  save_step: 500

# Input type
control_type: traj_pos #traj_vel or vel or pos or traj_pos

# Environment Parameters
running_step: 0.0010   # in seconds
door_frame_tolerances: 0.01 # 0.005
max_update_rate: 3000 # 1000 = x1 speed

force_limit: 300.000000  # In position control:300 ~ over 1000 (only rotate y2.5 ~ z-32)
torque_limit: 30.000000 # In position control:30 ~  (only rotate z1 ~ y-12)

obs_params:
  joint_n: 10 # 1/3.14 = 0.318471338
  eef_n: 10 # 1
  eef_rpy_n: 10 # 1
  force_n: 5 # 1
  torque_n: 5 # 1
  image_n: 0.0175 # 1/255 = 0.003921569

observations: # Observations you want to use for the learning
  - shp_joint_ang
  - shl_joint_ang
  - elb_joint_ang
  - wr1_joint_ang
  - wr2_joint_ang
  - wr3_joint_ang
#  - shp_joint_vel
#  - shl_joint_vel
#  - elb_joint_vel
#  - wr1_joint_vel
#  - wr2_joint_vel
#  - wr3_joint_vel
  - eef_x
  - eef_y
  - eef_z
  - eef_rpy_x
  - eef_rpy_y
  - eef_rpy_z
  - force_x
  - force_y
  - force_z
  - torque_x
  - torque_y
  - torque_z
  - image_data

joint_names: ['shoulder_link', 'upper_arm_link', 'forearm_link', 'wrist_1_link', 'wrist_2_link', 'wrist_3_link']

init_joint_pose0:
  shp: 0
  shl: -1.5707
  elb: 0
  wr1: -1.5707
  wr2: 0
  wr3: 0

init_joint_pose1:
  shp: 1.4410264033536153
  shl: -1.6883855417730604
  elb: 2.6771635884219136
  wr1: 2.1674968289716623
  wr2: -1.4199836765311638
  wr3: 1.5702616267031195

init_joint_pose2:
  shp: 1.488122534496775
  shl: -1.4496597816566892
  elb: 2.4377209990850974
  wr1: 2.168370898415174
  wr2: -1.4670589583209175
  wr3: 1.5709174034899702

init_grp_pose1:
  r_drive: 0
  l_drive: 0
  r_follower: 0
  l_follower: 0
  r_spring: 0
  l_spring: 0

init_grp_pose2:
  r_drive: 0.31
  l_drive: 0.31
  r_follower: -0.31
  l_follower: -0.31
  r_spring: 0.31
  l_spring: 0.31
# last time: 0.35NG 0.38, 0.4NG

joint_limits_array:
  shp_max: 3.14
  shp_min: -3.14
  shl_max: 3.14
  shl_min: -3.14
  elb_max: 3.14
  elb_min: -3.14
  wr1_max: 3.14
  wr1_min: -3.14
  wr2_max: 3.14
  wr2_min: -3.14
  wr3_max: 3.14
  wr3_min: -3.14


joint_velocity_limits_array:
  shp_max: 1.0
  shp_min: -1.0
  shl_max: 1.0
  shl_min: -1.0
  elb_max: 1.0
  elb_min: -1.0
  wr1_max: 1.0
  wr1_min: -1.0
  wr2_max: 1.0
  wr2_min: -1.0
  wr3_max: 1.0
  wr3_min: -1.0


#cartesian_limits:
#  x_max: 1.0
#  x_min: 1.0
#  y_max: -1.0
#  y_min: -1.0
#  z_max: -1.0
#  z_min: -1.0

episode_done_criteria:
  - panel_10deg

  #- minimum_height
  #- monoped_vertical_orientation
